javascripts/2048.js

```diff
@@ -1,32 +1,276 @@
+  tile_array[col][row] = tile;
+}
+
+
+// Creates 2 tiles at the start of the game
+function initializeGame() {
+  createTile();
+  createTile();
+}
+
+// Random number generated between 0 and 3
+function rando_num(){
+  return Math.floor(Math.random() * 4);
+}
+
+// See if a tile already exists at a particular column and row
+function checkLocation(column, row) {
+  if (tile_array[column][row] !== undefined) {
+    column = rando_num();
+    row = rando_num();
+    checkLocation(column, row);
+  }
+  return [column, row];
+}
+
+
+function createTile() {
+
+  // Check for empty spaces before creating
+
+    var new_column = rando_num();
+    var new_row = rando_num();
+
+    var result = checkLocation(new_column, new_row);
+
+    var column = result[0];
+    var row = result[1];
+
+    // Create new div element with tile class, location attributes, and value attribute
+    var newTile = $( '<div= class="tile">2</div>');
+    newTile.attr("data-col", 'c' + column);
+    newTile.attr("data-row", 'r' + row);
+    newTile.attr("data-val", "2");
+    $('#gameboard').append(newTile);
+
+    setTileInArray(column, row, newTile);
+    pop(newTile);
+}
+
+
+function tileCollide(tile, neighbor, direction) {
+  // tile should be checked against neighbor
+  //console.log('tile', tile, 'neighbor', neighbor);
+  if (tile.attr('data-val') == neighbor.attr('data-val')) {
+    //move tile to neighbors spot
+    var new_col = neighbor.attr('data-col');
+    var new_row = neighbor.attr('data-row');
+    tile.attr('data-col', new_col);
+    tile.attr('data-row', new_row);
+    //console.log(tile);
+    // take col and row and number and set for tile to be deleted
+    var col = +(new_col.slice(1));
+    var row = +(new_row.slice(1));
+
+    // reset tile array so updated tile is in correct spot
+    setTileInArray(col, row, tile);
+
+    // update tile
+    updateTile(tile);
+
+    // remove tile
+    setTimeout(function() {
+      deleteTile(neighbor);
+    }, 500);
+
+
+    // reset tile array so deleted spot is set to undefined
+    resetTileArray(col, row, direction)
+  }
+}
+
+  function deleteTile(neighbor){
+    junkTile = neighbor;
+    console.log("junk", junkTile)
+    $(junkTile).remove();
+  }
+
+
+  function resetTileArray(col, row, direction){
+    if (direction == 'up'){
+      var row = (row + 1);
+      setUndefinedTile(col, row);
+    } else if (direction == 'down') {
+      var row = (row - 1);
+      setUndefinedTile(col, row);
+    } else if (direction == 'left') {
+      var col = (col + 1);
+      setUndefinedTile(col, row);
+    } else if (direction == 'right') {
+      var col = (col - 1);
+      setUndefinedTile(col, row);
+    }
+  }
+
+ function updateTile(tile) {
   var new_tile_value = tile.attr("data-val") * 2;
   tile.attr("data-val", new_tile_value);
   tile.text(new_tile_value);
+ }
+
 
+function moveTile(direction) {
+  // needs conditionals to check for occupied grids spaces
   switch(direction) {
     case 38: //up
```

Comment:
To make things easier, instead of using numbers, you could define a global const object like so:
```
const DIRECTION = {
  up: 38,
  down: 40,
  left: 37,
  right: 39
};
```

And then you'd be able to check direction by doing this:
```
switch(direction) {
 case DIRECTION.up:
 case DIRECTION.down:
 ...
}
```